storage/tests/unit/test_blob.py

```diff
@@ -441,7 +441,9 @@ def _mock_requests_response(
         response.status_code = status_code
         response.headers.update(headers)
         if stream:
-            response.raw = io.BytesIO(content)
+            raw = io.BytesIO(content)
+            raw.headers = headers
+            response.raw = raw
```

Comment:
Why is `google-resumable-media` looking for headers on the response's stream, rather than the response itself?

Reply:
@tseaver Good point. Let's discuss, but the discussion shouldn't block the PR from un-breaking tests for everyone? (Will follow up with another comment.)

Reply:
The "why?" is "because that's what `urllib3` [does][1]". The [change][2] that added it actually monkey-patches the `gzip` decoder attached to `response.raw`, so I used the same logic as `urllib3` uses in `urllib3.response.HTTPResponse._init_decoder`.

But, as you pointed out, `urrlib3` **can't** assume it's being used along-side / used by `requests` but we **can** assume that, so we **could** do the `content-encoding` header check from `requests.models.Response.headers`.

[1]: https://github.com/shazow/urllib3/blob/3a74b5b7862a1b55c3a059e9621aea4322e786f8/urllib3/response.py#L250-L258
[2]: https://github.com/GoogleCloudPlatform/google-resumable-media-python/pull/36